substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/monitor/OwnedReentrantLock.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2022, 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2022, 2022, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.core.monitor;
+
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.graalvm.nativeimage.CurrentIsolate;
+import com.oracle.svm.core.jfr.SubstrateJVM;
+
+public class OwnedReentrantLock extends ReentrantLock {
+    private long prevOwnerTid;
+    private long currOwnerTid;
+
+    public long getPrevOwnerTid() {
+        return prevOwnerTid;
+    }
+
+    public OwnedReentrantLock() {
+        super();
+        currOwnerTid = SubstrateJVM.get().getThreadId(CurrentIsolate.getCurrentThread());
+    }
+
+    public boolean tryLockOwnedLock() {
```

Comment:
You could rename this method to `enter` or `monitorenter`.
If you move the call to `JavaMonitorEnterEvent.emit(...);` into this method as well, then it should be sufficient to store the latest owner in a single field.

Reply:
renamed and moved call to emit event